gm_dotnet_managed/GmodNET.API/ModuleAssemblyLoadContext.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Runtime.Loader;
+
+namespace GmodNET.API
+{
+    /// <summary>
+    /// An extension of <see cref="AssemblyLoadContext" /> class with GmodNET specific features.
+    /// </summary>
+    public abstract class ModuleAssemblyLoadContext : AssemblyLoadContext
+    {
+        /// <summary>
+        /// Get associated module name.
+        /// </summary>
+        public abstract string ModuleName {get; }
```

Comment:
Could we potentially use a virtual here so that inheritors don't have to create a backing field and then can just set in the constructor?

Reply:
Yeah. makes sense. Let me fix it now.

Reply:
@AaronJThompson Fixed with the last commit.

Reply:
Awesome! Should be easier to use